go/src/koding/klientctl/auth/register.go

```diff
@@ -0,0 +1,187 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"koding/klientctl/ctlcli"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/koding/logging"
+)
+
+// NewRegisterSubCommand provides the subcommand for registering
+func NewRegisterSubCommand(log logging.Logger) cli.Command {
+	return cli.Command{
+		Name:   "register",
+		Usage:  "Registers the user",
+		Action: ctlcli.ExitAction(RegisterCommand, log, "register"),
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "username",
+				Usage: "Username to register",
+			},
+			cli.StringFlag{
+				Name:  "firstName",
+				Usage: "firstName to register",
+			},
+			cli.StringFlag{
+				Name:  "lastName",
+				Usage: "lastName to register",
+			},
+			cli.StringFlag{
+				Name:  "password",
+				Usage: "Username password to register",
+			},
+			cli.StringFlag{
+				Name:  "email",
+				Usage: "Email Address to register",
+			},
+			cli.StringFlag{
+				Name:  "team",
+				Usage: "Team Name to register",
+			},
+			cli.StringFlag{
+				Name:  "company",
+				Usage: "Team Company Name to register (default: team name)",
+			},
+			cli.BoolFlag{
+				Name:  "newsletter",
+				Usage: "Do you want to get occasional newsletters from Koding?",
+			},
+			cli.BoolFlag{
+				Name:  "alreadyMember",
+				Usage: "Do you already registered to Koding?",
+			},
+		},
+	}
+}
+
+// RegisterRequest holds the registration request to Koding.
+type RegisterRequest struct {
+	Username      string `json:"username"`
+	Password      string `json:"password"`
+	Email         string `json:"email"`
+	Slug          string `json:"slug"`
+	Newsletter    bool   `json:"newsletter,string"`
+	AlreadyMember bool   `json:"alreadyMember,string"`
+}
+
+// RegisterCommand displays version information like Environment or Kite Query ID.
+func RegisterCommand(c *cli.Context, log logging.Logger, _ string) int {
+	if len(c.Args()) != 2 {
+		cli.ShowCommandHelp(c, "register")
+		return 1
+	}
+	host := c.GlobalString("host")
+
+	// TODO(mehmetali): make a generalized client to be used in klientctl
+	client := &http.Client{}
+
+	token, err := doRegisterRequest(c, client, host)
+	if err != nil {
+		log.Error(err.Error())
```

Comment:
Please use `log.Error("%s", err)` - if err.Error() string contained `%` character, the whole log line will be broken when used like that: `log.Error(err.Error())`.

Also we should print some info to the user, like: `fmt.Fprintln(os.Stderr, "Register failed with error:", err)`. The `log` logs to a file (think of it like diagnostics) - e.g. under macOS it's /Library/Logs/kd.log.

Yes, the logging handling is really verbose in kd currently - we plan to rework it in future by changing cli framework.

Currently we're handling